services/cards-consultation/src/main/java/org/opfab/cards/consultation/services/CardSubscriptionService.java

```diff
@@ -93,6 +101,34 @@ private void sendHeartbeatMessageInAllSubscriptions()
                     sub.publishDataIntoSubscription("HEARTBEAT");
                 }
             });
+        }
+    }
+
+    private void compareLastHeartbeatReceived()
+    {
+        for(;;)
+        {
+            try
+                {
+                Thread.sleep(heartbeatDelay);
+                }
+                catch (InterruptedException ex)
+                {
+                    log.error("Impossible to read receiving heartbeats",ex);
+                    Thread.currentThread().interrupt(); // Cf sonar java:S2142 "InterruptedException" should not be ignored
+                    return; 
+                }
+            log.debug("Comparing last heartbeat received");
+            cache.keySet().forEach(key -> {
+                CardSubscription sub = cache.get(key); 
+                if (sub != null) // subscription can be null if it has been evicted during the process of going throw the keys
+                {
+                    log.debug("Time since last heart beat : {}", Instant.now().getEpochSecond() - sub.getHeartbeatReceptionDate().getEpochSecond());
+                    if (Instant.now().getEpochSecond() - sub.getHeartbeatReceptionDate().getEpochSecond()  > 150) {
```

Comment:
Use a configuration value instead of 150  

Name : heardbeatDelayInSecondsToConsiderUserDisconnected
Default value : 100